cmd/move_labels/move_labels.go

```diff
@@ -114,6 +114,9 @@ func moveLabelsInDir(c *configuration) ([]*build.File, error) {
 
 func moveLabelsInFile(file *build.File, from, to string) {
 	build.Edit(file, func(x build.Expr, _ []build.Expr) build.Expr {
+		if loadStmt, ok := x.(*build.LoadStmt); ok {
```

Comment:
This special handling is inside the function that build.Edit is recursively applying on the expressions. We were previously only handling expressions of type StringExpr, now we need to explicitly handle LoadStmt as well.

Reply:
`build.Edit` is supposed to visit every node in the AST root at its argument. I looked at the implementation, and it doesn't seem like it actually has a case for `LoadStmt`. I think that's a bug that should be fixed in that library.

Could you add a `TODO` here explaining why the special case here is needed?

Reply:
build.Edit is working as expected. If you notice the original function that we are providing to build.Edit, _we_ skip everything that does not type assert to `*build.StringExpr`. So it is on us to add one more type assert to consider LoadStmt too.

Reply:
I don't think we need to transform `LoadStmt` itself though; we need to transform `Module *StringExpr` inside `*LoadStmt`.

The purpose of this function is to transform all the labels that appear anywhere in a file. All labels have type `*StringExpr`, so we should be able to call `build.Edit`, check whether the AST node being visited is a `*StringExpr`, and then return a replacement `*StringExpr` if we actually make a change.

We shouldn't need a special case for `*LoadStmt`, just as we shouldn't need a special case for `*CallExpr`. `build.Edit` is supposed to visit through all child nodes, so our callback should get called on the `Module` field of each `*LoadStmt`.

I think the bug is that `build.WalkOnce`, which is used in the implementation of `build.Edit`, [does not have a case](https://github.com/bazelbuild/buildtools/blob/9f8fdb20dd423621ef00ced33dcb40204703c2c8/build/walk.go#L74) for `*LoadStmt`. It visits children of other AST types, but not that one.

I'll send a PR to fix that if I have time this afternoon, but until it's fixed, I think there should be a TODO comment here explaining why the special case is necessary.

Reply:
Aah! I apologize. I remember having made this remark myself when I first came across it. But I forgot since then, and caused the confusion just now. Now added a TODO.